flit/upload.py

```diff
@@ -237,10 +239,10 @@ def upload_file(file:Path, metadata:Metadata, repo):
     resp.raise_for_status()
 
 
-def do_upload(file:Path, metadata:Metadata, repo_name=None):
+def do_upload(file:Path, metadata:Metadata, pypirc_path:Path, repo_name=None):
```

Comment:
```suggestion
def do_upload(file:Path, metadata:Metadata, pypirc_path, repo_name=None):
```

I don't see anything converting the string to a Path object, and I'm not sure `get_repositories` would work if it was.